include/kodlab_mjbots_sdk/behavior_manager.h

```diff
@@ -83,93 +98,105 @@ class BehaviorManager
-      if (!behaviors_[next_idx_]->is_initialized())
-      {
+      if (!behaviors_[next_idx_]->is_initialized()) {
         behaviors_[next_idx_]->Init();
       }
       behaviors_[selected_idx]->Stop(*behaviors_[next_idx_]); // stop current behavior
       switching_behaviors_ = true;  // indicate transition is in progress
-    } else
-    {
+    } else {
       LOG_WARN("Invalid behavior selected: %d", next_idx);
     }
   }
 
   /**
-   * @brief Constructs and initializes a behavior
-   * @tparam BehaviorType `kodlab::Behavior`-derived type
-   * @tparam ConstructorArgs constructor arguments
-   * @param args behavior constructor arguments
-   * @return unique pointer to the behavior if initialization was successful,
-   *         nullptr otherwise
+   * @brief Set the current behavior
+   * @param name name of behavior in `behaviors_`
    */
-  template<class BehaviorType, typename... ConstructorArgs>
-  std::unique_ptr<BehaviorType> InitBehavior(ConstructorArgs &&... args)
-  {
-    static_assert(std::is_base_of<kodlab::Behavior<Robot>, BehaviorType>::value,
-                  "BehaviorType must be a `kodlab::Behavior`-derived type");
-    auto b = std::make_unique<BehaviorType>(args...);
-    if (b->Init())
-    {
-      b->set_initialized();
-      return std::move(b);
-    } else { return nullptr; }
+  virtual void SetBehavior(const std::string &name) {
+    if (name == KILL_BEHAVIOR_NAME) { SetBehavior(KILL_ROBOT_IDX); }
+    else {
+      auto idx = get_behavior_index(name);
+      if (idx >= 0) { SetBehavior(idx); }
+      else {
+        LOG_WARN("Invalid behavior selected: %s at index %d",
+                 name.c_str(),
+                 idx);
+      }
+    }
   }
 
   /**
    * @brief Add a behavior to the behaviors list and initialize it
    * @tparam BehaviorType `kodlab::Behavior`-derived type
    * @tparam ConstructorArgs constructor arguments
    * @param args behavior constructor arguments
+   * @return index of behavior in behaviors list if added successfully, -1
+   * otherwise
    */
   template<class BehaviorType, typename... ConstructorArgs>
-  void AddBehavior(ConstructorArgs &&... args)
-  {
+  int AddBehavior(ConstructorArgs &&... args) {
     static_assert(std::is_base_of<kodlab::Behavior<Robot>, BehaviorType>::value,
                   "BehaviorType must be a `kodlab::Behavior`-derived type");
-    auto b = InitBehavior<BehaviorType>(args...);
-    if (b)
-    {
-      names_.push_back(b->get_name());
+    auto b = ConstructBehavior<BehaviorType>(args...);
+    if (b) {
+      int idx = behaviors_.size();
+      std::string name = b->get_name();
+      names_.push_back(name);
+      name_to_idx_map_.emplace(std::make_pair(name, idx));
       behaviors_.emplace_back(std::move(b));
-    } else
-    {
+      return idx;
+    } else {
       LOG_WARN(
           "%s behavior could not be initialized and will not be added to behaviors list.",
           b->get_name().c_str());
+      return -1;
```

Comment:
hmm. Thats probably bad since -1 is the kill behavior index. lets make it an optional int.

Reply:
Haha probably for the best, fixed in 6bc58c22c7bfe4d156e3962308c3fbf73fc639c6